ui/browser/actions/main-actions.js

```diff
@@ -42,3 +42,7 @@ export function setCurrentTab(pageIndex) {
 export function setPageOrder(pageOrder) {
   return { type: types.SET_PAGE_ORDER, pageOrder };
 }
+
+export function togglePagesSection(opened) {
+  return { type: types.TOGGLE_PAGES_SECTION, opened };
```

Comment:
I feel like you're straddling two action models.  On the one hand, you want to say "toggled"; on the other, you want to say "set to ...".  You have some code checking for `undefined` later to handle the former, which feels wrong to me.

Is there value supporting both verbs?   With one action?  In general, I prefer to have actions that are facts: "opened pages section", "closed pages section", and therefore don't rely on previous state to interpret.  Is that not a Redux-y thing to do?